cmd/scale.go

```diff
@@ -229,8 +229,8 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 				continue
 			}
 
-			osPublisher := *vm.StorageProfile.ImageReference.Publisher
-			if strings.EqualFold(osPublisher, "MicrosoftWindowsServer") {
+			osPublisher := vm.StorageProfile.ImageReference.Publisher
```

Comment:
Won't this be a nil pointer dereference if StorageProfile and/or ImageReference are nil?

Reply:
It would, although OTOH these have just been returned by an Azure API query for a running VM therefore I was supposing this scenario would be improbable.

Reply:
it would be possible if the VM returned doesn't have those fields, I thought that was the intent of this PR? Do OpenShift VMs have and ImageReference but no osPublisher?

Reply:
Correct, VMs that are not provisioned by the marketplace have vm.StorageProfile.ImageReference.ID set, but vm.StorageProfile.ImageReference.Publisher is nil.

Reply:
ok great, thanks for clarifying